File: packages/ruviz/examples/histogram.ts
import { createPlot } from "../src/index.js";
import { sampleDistribution } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "histogram",
  title: "Histogram",
  summary: "A distribution view built from a deterministic sample.",
  category: "statistical",
  createPlot() {
    return createPlot()
      .sizePx(640, 320)
      .title("Histogram")
      .xlabel("value")
      .histogram(sampleDistribution());
  },
};

File: bindings/python/examples/histogram.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, sample_distribution, save_example

META = ExampleMeta(
    slug="histogram",
    title="Histogram",
    summary="A distribution view built from a deterministic sample with an explicit bin count.",
    section="Statistical plots",
)


def build_plot():
    return (
        base_plot("Histogram")
        .xlabel("value")
        .histogram(sample_distribution(), bins=24, color="#f97316", alpha=0.85)
    )


if __name__ == "__main__":
    save_example(META, build_plot())
